x-pack/platform/plugins/private/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/indices/flyout/steps/warning/warning_step.tsx

```diff
@@ -92,27 +97,50 @@ export const WarningsFlyoutStep: React.FunctionComponent<WarningsConfirmationFly
   return (
     <>
       <EuiFlyoutBody>
+        {meta.isFrozen && <FrozenCallOut />}
         {warnings.length > 0 && (
           <>
-            <EuiCallOut
-              title={
-                <FormattedMessage
-                  id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.warningsStep.destructiveCallout.calloutTitle"
-                  defaultMessage="This index requires destructive changes that cannot be reversed"
-                />
-              }
-              color="warning"
-              iconType="warning"
-            >
-              <p>
-                <FormattedMessage
-                  id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.warningsStep.destructiveCallout.calloutDetail"
-                  defaultMessage="Back up the index before continuing. To proceed with the reindex, accept each change."
-                />
-              </p>
-            </EuiCallOut>
+            {flow === 'reindex' && (
+              <EuiCallOut
+                title={
+                  <FormattedMessage
+                    id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.warningsStep.reindex.calloutTitle"
+                    defaultMessage="This index requires destructive changes that cannot be reversed"
+                  />
+                }
+                color="warning"
+                iconType="warning"
+              >
+                <p>
+                  <FormattedMessage
+                    id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.warningsStep.reindex.calloutDetail"
+                    defaultMessage="Back up the index before continuing. To proceed with the reindex, accept each change."
+                  />
+                </p>
+              </EuiCallOut>
+            )}
+            {flow === 'readonly' && (
+              <EuiCallOut
+                title={
+                  <FormattedMessage
+                    id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.warningsStep.readonly.calloutTitle"
+                    defaultMessage="Marking the index as read-only will cause future update operations to fail"
+                  />
+                }
+                color="warning"
+                iconType="warning"
+              >
+                <p>
+                  <FormattedMessage
+                    id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.warningsStep.readonly.calloutDetail"
+                    defaultMessage="Please make sure that you no longer need to write or update documents in this index anymore. You will not be able to revert the read-only status after upgrading, as it is a pre-requisite for N-2 compatibility. If unsure, please consider reindexing this index instead."
```

Comment:
I'm not entirely sure this is correct: by making this index read only now we are essentially deferring reindexing work if we ever need to write to this index again. In 9.0 we can reindex this index to a new one to regain the ability to write to this data.

We should also avoid language like `N-2`.

Reply:
Makes sense, we don't want to discourage users from going for this option.
How would you phrase it?